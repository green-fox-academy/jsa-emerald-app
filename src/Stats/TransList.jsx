import React from 'react';
import { View, Text } from 'react-native';
import { ListItem } from 'react-native-elements';
import { useSelector } from 'react-redux';
import GradientIcon from '../Common/GradientIcon';
import utils from './utils';
import setThemeStyle from '../Common/theme/setThemeStyle';

export default function TransList(props) {
  const { transactions } = props;
  const { themeMode } = useSelector((state) => state.theme);
  const styles = setThemeStyle(themeMode);
  return (
    transactions.map((groupedRecordsByDate, index) => (
      <View style={styles.card} key={`DateGroup:${index + 1}`}>
        <View style={styles.cardAlign}>
          <Text style={styles.cardHeader}>
            {groupedRecordsByDate[0].date}
          </Text>
          <Text style={styles.cardHeader}>
            {utils.sumAmount(groupedRecordsByDate)}
          </Text>
        </View>
        {groupedRecordsByDate.map((item, idx) => (
          <ListItem
            key={`InfoGroup:${idx + 1}`}
            leftElement={(
              <GradientIcon
                name={item.labelName ? item.labelName.icon : 'home'}
                color={item.labelName && item.labelName.color ? item.labelName.color : '#9e87fc'}
                iconFamily={item.labelName && item.labelName.iconFamily ? item.labelName.iconFamily : ''}
              />
            )}
            title={item.labelName ? item.labelName.name : 'Unknown'}
            titleStyle={styles.listHeading}
            subtitle={item.creator ? item.creator : 'Unknown'}
            subtitleStyle={item.creator ? { color: 'grey' } : { display: 'none' }}
            rightTitle={utils.transType(item.amount, item.type)}
            rightTitleStyle={styles.listHeading}
            containerStyle={{ backgroundColor: 'rgba(255,0, 0, 0)' }}
          />
        ))}
      </View>
    ))
  );
}
