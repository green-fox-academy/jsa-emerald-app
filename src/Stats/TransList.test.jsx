import React from 'react';
import renderer from 'react-test-renderer';
import TransList from './TransList';

describe('<TransList />', () => {
  it('should render with selected label', () => {
    const dataList = [
      [
        {
          amount: 111.00,
          date: 'December 24th 2019',
          label: {
            color: '#aa78db',
            icon: 'shoppingcart',
            iconFamily: 'antdesign',
            name: 'Shopping',
          },
          comment: 'expand',
          type: 'Expense',
        },
      ],
    ];
    const tree = renderer.create(<TransList
      transactions={dataList}
    />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render with default label', () => {
    const dataList = [
      [
        {
          amount: 111.00,
          date: 'December 24th 2019',
          type: 'Expense',
        },
      ],
    ];
    const tree = renderer.create(<TransList
      transactions={dataList}
    />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
