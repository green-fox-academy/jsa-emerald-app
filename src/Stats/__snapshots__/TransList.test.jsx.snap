// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TransList /> render 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "borderRadius": 5,
      "elevation": 2,
      "marginTop": 20,
      "padding": 10,
      "shadowColor": "grey",
      "shadowOffset": Object {
        "height": 3,
        "width": 2,
      },
      "shadowOpacity": 0.08,
      "shadowRadius": 5,
    }
  }
>
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#777777",
          "fontSize": 18,
          "margin": 10,
        }
      }
    >
      December 24th 2019
    </Text>
    <Text
      style={
        Object {
          "color": "#777777",
          "fontSize": 18,
          "margin": 10,
        }
      }
    >
      -$111
    </Text>
  </View>
  <View>
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "white",
          "borderColor": "#bcbbc1",
          "flexDirection": "row",
          "padding": 14,
        }
      }
    >
      <ViewManagerAdapter_ExpoLinearGradient
        proxiedProperties={
          Object {
            "colors": Array [
              4289362139,
              4294945023,
            ],
            "endPoint": Array [
              0.9,
              0.2,
            ],
            "locations": undefined,
            "startPoint": Array [
              0.1,
              0.9,
            ],
          }
        }
        style={
          Object {
            "borderRadius": 23,
            "padding": 8,
          }
        }
      >
        <View>
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "transparent",
                "justifyContent": "center",
              }
            }
            theme={
              Object {
                "colors": Object {
                  "disabled": "hsl(208, 8%, 90%)",
                  "divider": "#bcbbc1",
                  "error": "#ff190c",
                  "grey0": "#393e42",
                  "grey1": "#43484d",
                  "grey2": "#5e6977",
                  "grey3": "#86939e",
                  "grey4": "#bdc6cf",
                  "grey5": "#e1e8ee",
                  "greyOutline": "#bbb",
                  "platform": Object {
                    "android": Object {
                      "error": "#f44336",
                      "primary": "#2196f3",
                      "secondary": "#9C27B0",
                      "success": "#4caf50",
                      "warning": "#ffeb3b",
                    },
                    "ios": Object {
                      "error": "#ff3b30",
                      "primary": "#007aff",
                      "secondary": "#5856d6",
                      "success": "#4cd964",
                      "warning": "#ffcc00",
                    },
                  },
                  "primary": "#2089dc",
                  "searchBg": "#303337",
                  "secondary": "#8F0CE8",
                  "success": "#52c41a",
                  "warning": "#faad14",
                },
              }
            }
            underlayColor="white"
          >
            <Text />
          </View>
        </View>
      </ViewManagerAdapter_ExpoLinearGradient>
      <View
        width={16}
      />
      <View
        style={
          Object {
            "flex": 1,
            "justifyContent": "center",
          }
        }
      >
        <Text
          style={
            Object {
              "backgroundColor": "transparent",
              "fontSize": 17,
            }
          }
          testID="listItemTitle"
          theme={
            Object {
              "colors": Object {
                "disabled": "hsl(208, 8%, 90%)",
                "divider": "#bcbbc1",
                "error": "#ff190c",
                "grey0": "#393e42",
                "grey1": "#43484d",
                "grey2": "#5e6977",
                "grey3": "#86939e",
                "grey4": "#bdc6cf",
                "grey5": "#e1e8ee",
                "greyOutline": "#bbb",
                "platform": Object {
                  "android": Object {
                    "error": "#f44336",
                    "primary": "#2196f3",
                    "secondary": "#9C27B0",
                    "success": "#4caf50",
                    "warning": "#ffeb3b",
                  },
                  "ios": Object {
                    "error": "#ff3b30",
                    "primary": "#007aff",
                    "secondary": "#5856d6",
                    "success": "#4cd964",
                    "warning": "#ffcc00",
                  },
                },
                "primary": "#2089dc",
                "searchBg": "#303337",
                "secondary": "#8F0CE8",
                "success": "#52c41a",
                "warning": "#faad14",
              },
            }
          }
        >
          Shopping
        </Text>
        <Text
          style={
            Object {
              "backgroundColor": "transparent",
              "color": "grey",
              "fontSize": 15,
            }
          }
          theme={
            Object {
              "colors": Object {
                "disabled": "hsl(208, 8%, 90%)",
                "divider": "#bcbbc1",
                "error": "#ff190c",
                "grey0": "#393e42",
                "grey1": "#43484d",
                "grey2": "#5e6977",
                "grey3": "#86939e",
                "grey4": "#bdc6cf",
                "grey5": "#e1e8ee",
                "greyOutline": "#bbb",
                "platform": Object {
                  "android": Object {
                    "error": "#f44336",
                    "primary": "#2196f3",
                    "secondary": "#9C27B0",
                    "success": "#4caf50",
                    "warning": "#ffeb3b",
                  },
                  "ios": Object {
                    "error": "#ff3b30",
                    "primary": "#007aff",
                    "secondary": "#5856d6",
                    "success": "#4cd964",
                    "warning": "#ffcc00",
                  },
                },
                "primary": "#2089dc",
                "searchBg": "#303337",
                "secondary": "#8F0CE8",
                "success": "#52c41a",
                "warning": "#faad14",
              },
            }
          }
        >
          expand
        </Text>
      </View>
      <View
        width={16}
      />
      <View
        style={
          Object {
            "alignItems": "flex-end",
            "flex": 0.5,
            "justifyContent": "center",
          }
        }
      >
        <Text
          style={
            Object {
              "backgroundColor": "transparent",
              "color": "rgba(0, 0, 0, 0.54)",
              "fontSize": 17,
            }
          }
          theme={
            Object {
              "colors": Object {
                "disabled": "hsl(208, 8%, 90%)",
                "divider": "#bcbbc1",
                "error": "#ff190c",
                "grey0": "#393e42",
                "grey1": "#43484d",
                "grey2": "#5e6977",
                "grey3": "#86939e",
                "grey4": "#bdc6cf",
                "grey5": "#e1e8ee",
                "greyOutline": "#bbb",
                "platform": Object {
                  "android": Object {
                    "error": "#f44336",
                    "primary": "#2196f3",
                    "secondary": "#9C27B0",
                    "success": "#4caf50",
                    "warning": "#ffeb3b",
                  },
                  "ios": Object {
                    "error": "#ff3b30",
                    "primary": "#007aff",
                    "secondary": "#5856d6",
                    "success": "#4cd964",
                    "warning": "#ffcc00",
                  },
                },
                "primary": "#2089dc",
                "searchBg": "#303337",
                "secondary": "#8F0CE8",
                "success": "#52c41a",
                "warning": "#faad14",
              },
            }
          }
        >
          -$111
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`<TransList /> render 2`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "borderRadius": 5,
      "elevation": 2,
      "marginTop": 20,
      "padding": 10,
      "shadowColor": "grey",
      "shadowOffset": Object {
        "height": 3,
        "width": 2,
      },
      "shadowOpacity": 0.08,
      "shadowRadius": 5,
    }
  }
>
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#777777",
          "fontSize": 18,
          "margin": 10,
        }
      }
    >
      December 24th 2019
    </Text>
    <Text
      style={
        Object {
          "color": "#777777",
          "fontSize": 18,
          "margin": 10,
        }
      }
    >
      -$111
    </Text>
  </View>
  <View>
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "white",
          "borderColor": "#bcbbc1",
          "flexDirection": "row",
          "padding": 14,
        }
      }
    >
      <ViewManagerAdapter_ExpoLinearGradient
        proxiedProperties={
          Object {
            "colors": Array [
              4288579580,
              4294950399,
            ],
            "endPoint": Array [
              0.9,
              0.2,
            ],
            "locations": undefined,
            "startPoint": Array [
              0.1,
              0.9,
            ],
          }
        }
        style={
          Object {
            "borderRadius": 23,
            "padding": 8,
          }
        }
      >
        <View>
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "transparent",
                "justifyContent": "center",
              }
            }
            theme={
              Object {
                "colors": Object {
                  "disabled": "hsl(208, 8%, 90%)",
                  "divider": "#bcbbc1",
                  "error": "#ff190c",
                  "grey0": "#393e42",
                  "grey1": "#43484d",
                  "grey2": "#5e6977",
                  "grey3": "#86939e",
                  "grey4": "#bdc6cf",
                  "grey5": "#e1e8ee",
                  "greyOutline": "#bbb",
                  "platform": Object {
                    "android": Object {
                      "error": "#f44336",
                      "primary": "#2196f3",
                      "secondary": "#9C27B0",
                      "success": "#4caf50",
                      "warning": "#ffeb3b",
                    },
                    "ios": Object {
                      "error": "#ff3b30",
                      "primary": "#007aff",
                      "secondary": "#5856d6",
                      "success": "#4cd964",
                      "warning": "#ffcc00",
                    },
                  },
                  "primary": "#2089dc",
                  "searchBg": "#303337",
                  "secondary": "#8F0CE8",
                  "success": "#52c41a",
                  "warning": "#faad14",
                },
              }
            }
            underlayColor="white"
          >
            <Text />
          </View>
        </View>
      </ViewManagerAdapter_ExpoLinearGradient>
      <View
        width={16}
      />
      <View
        style={
          Object {
            "flex": 1,
            "justifyContent": "center",
          }
        }
      >
        <Text
          style={
            Object {
              "backgroundColor": "transparent",
              "fontSize": 17,
            }
          }
          testID="listItemTitle"
          theme={
            Object {
              "colors": Object {
                "disabled": "hsl(208, 8%, 90%)",
                "divider": "#bcbbc1",
                "error": "#ff190c",
                "grey0": "#393e42",
                "grey1": "#43484d",
                "grey2": "#5e6977",
                "grey3": "#86939e",
                "grey4": "#bdc6cf",
                "grey5": "#e1e8ee",
                "greyOutline": "#bbb",
                "platform": Object {
                  "android": Object {
                    "error": "#f44336",
                    "primary": "#2196f3",
                    "secondary": "#9C27B0",
                    "success": "#4caf50",
                    "warning": "#ffeb3b",
                  },
                  "ios": Object {
                    "error": "#ff3b30",
                    "primary": "#007aff",
                    "secondary": "#5856d6",
                    "success": "#4cd964",
                    "warning": "#ffcc00",
                  },
                },
                "primary": "#2089dc",
                "searchBg": "#303337",
                "secondary": "#8F0CE8",
                "success": "#52c41a",
                "warning": "#faad14",
              },
            }
          }
        >
          Unknown
        </Text>
        <Text
          style={
            Object {
              "backgroundColor": "transparent",
              "color": "grey",
              "fontSize": 15,
            }
          }
          theme={
            Object {
              "colors": Object {
                "disabled": "hsl(208, 8%, 90%)",
                "divider": "#bcbbc1",
                "error": "#ff190c",
                "grey0": "#393e42",
                "grey1": "#43484d",
                "grey2": "#5e6977",
                "grey3": "#86939e",
                "grey4": "#bdc6cf",
                "grey5": "#e1e8ee",
                "greyOutline": "#bbb",
                "platform": Object {
                  "android": Object {
                    "error": "#f44336",
                    "primary": "#2196f3",
                    "secondary": "#9C27B0",
                    "success": "#4caf50",
                    "warning": "#ffeb3b",
                  },
                  "ios": Object {
                    "error": "#ff3b30",
                    "primary": "#007aff",
                    "secondary": "#5856d6",
                    "success": "#4cd964",
                    "warning": "#ffcc00",
                  },
                },
                "primary": "#2089dc",
                "searchBg": "#303337",
                "secondary": "#8F0CE8",
                "success": "#52c41a",
                "warning": "#faad14",
              },
            }
          }
        >
          Unknown
        </Text>
      </View>
      <View
        width={16}
      />
      <View
        style={
          Object {
            "alignItems": "flex-end",
            "flex": 0.5,
            "justifyContent": "center",
          }
        }
      >
        <Text
          style={
            Object {
              "backgroundColor": "transparent",
              "color": "rgba(0, 0, 0, 0.54)",
              "fontSize": 17,
            }
          }
          theme={
            Object {
              "colors": Object {
                "disabled": "hsl(208, 8%, 90%)",
                "divider": "#bcbbc1",
                "error": "#ff190c",
                "grey0": "#393e42",
                "grey1": "#43484d",
                "grey2": "#5e6977",
                "grey3": "#86939e",
                "grey4": "#bdc6cf",
                "grey5": "#e1e8ee",
                "greyOutline": "#bbb",
                "platform": Object {
                  "android": Object {
                    "error": "#f44336",
                    "primary": "#2196f3",
                    "secondary": "#9C27B0",
                    "success": "#4caf50",
                    "warning": "#ffeb3b",
                  },
                  "ios": Object {
                    "error": "#ff3b30",
                    "primary": "#007aff",
                    "secondary": "#5856d6",
                    "success": "#4cd964",
                    "warning": "#ffcc00",
                  },
                },
                "primary": "#2089dc",
                "searchBg": "#303337",
                "secondary": "#8F0CE8",
                "success": "#52c41a",
                "warning": "#faad14",
              },
            }
          }
        >
          -$111
        </Text>
      </View>
    </View>
  </View>
</View>
`;
